net/modemmanager/files/modemmanager.common

```diff
@@ -0,0 +1,323 @@
+	local action=$1
+	local sysfspath=$2
+
+	local parent_sysfspath status
+
+	parent_sysfspath=$(mm_find_physdev_sysfs_path $sysfspath)
+	[ -n "${parent_sysfspath}" ] || {
+		mm_log "error: parent device sysfspath not found"
+		return
+	}
+
+	status=$(mm_get_modem_wait_status "${parent_sysfspath}")
+
+	[ "$action" = "add" ] && {
+		case "${status}" in
+			"")
+				local cfg
+
+				cfg=$(mm_get_modem_config "${parent_sysfspath}")
+				if [ -n "${cfg}" ]; then
+					mm_log "interface '${cfg}' is set to configure device '${parent_sysfspath}'"
+					mm_log "now waiting for modem at sysfs path ${parent_sysfspath}"
+					mm_set_modem_wait_status "${parent_sysfspath}" "processed"
+					# Launch subshell for the explicit wait
+					( mm_wait_for_modem "${cfg}" "${parent_sysfspath}" ) > /dev/null 2>&1 &
+				else
+					mm_log "no need to wait for modem at sysfs path ${parent_sysfspath}"
+					mm_set_modem_wait_status "${parent_sysfspath}" "ignored"
+				fi
+				;;
+			"processed")
+				mm_log "already waiting for modem at sysfs path ${parent_sysfspath}"
+				;;
+			"ignored")
+				;;
+			*)
+				mm_log "error: unknown status read for device at sysfs path ${parent_sysfspath}"
+				;;
+		esac
+		return
+	}
+
+	[ "$action" = "remove" ] && {
+		local cfg
+
+		[ -n "$status" ] && {
+			local cfg
+
+			mm_log "cleanup wait for modem at sysfs path ${parent_sysfspath}"
+			mm_set_modem_wait_status "${parent_sysfspath}" ""
+
+			cfg=$(mm_get_modem_config "${parent_sysfspath}")
+			[ -n "${cfg}" ] && {
+				mm_log "setting interface '$cfg' as unavailable"
+				proto_set_available "${cfg}" 0
+			}
+		}
+		return
+	}
+}
+
+################################################################################
+# Cleanup interfaces
+
+mm_cleanup_interface_cb() {
+	local cfg="$1"
+
+	local proto
+	config_get proto "${cfg}" proto
+	[ "${proto}" = modemmanager ] || return 0
+
+	proto_set_available "${cfg}" 0
+}
+
+mm_cleanup_interfaces() {
+	config_load network
+	config_foreach mm_cleanup_interface_cb interface
+}
+
+################################################################################
+# Event reporting
+
+# Receives as input the action, the device name and the subsystem
+mm_report_event() {
+	local action="$1"
+	local name="$2"
+	local subsystem="$3"
+	local sysfspath="$4"
+
+	# Track/untrack events in cache
+	case "${action}" in
+		"add")
+			# On add events, store event details in cache (if not exists yet)
+			grep -qs "${name},${subsystem}" "${MODEMMANAGER_EVENTS_CACHE}" || \
+				echo "${action},${name},${subsystem},${sysfspath}" >> "${MODEMMANAGER_EVENTS_CACHE}"
+			;;
+		"remove")
+			# On remove events, remove old events from cache (match by subsystem+name)
+			sed -i "/${name},${subsystem}/d" "${MODEMMANAGER_EVENTS_CACHE}"
+			;;
+	esac
+
+	# Report the event
+	mm_log "event reported: action=${action}, name=${name}, subsystem=${subsystem}"
+	/usr/bin/mmcli --report-kernel-event="action=${action},name=${name},subsystem=${subsystem}" 1>/dev/null 2>&1 &
```

Comment:
Absolute path required here for mmcli?

Reply:
Hi, thanks for your review.  I am working on implementing your suggestions.  What would you suggest instead of `/usr/bin/mmcli` ?

Reply:
I would expect that mmcli is available via PATH? Then just `mmcli` should work...